### PR TITLE
fix(snap): remove support-logging

### PR DESCRIPTION
**Pull request by tonyespy**
Remove deprecated support-logging service from snap.

Signed-off-by: Tony Espy <espy@canonical.com>

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit subject follows the [Conventional Commits spec](https://github.com/zeke/semantic-pull-requests)
- [x] The commit message follows the [EdgeX Contributor Guide](https://wiki.edgexfoundry.org/display/FA/Contributor%27s+Guide)
- [NA] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [NA] `make test` has completed successfully

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [x] Other... Please describe: removes the deprecated support-logging service from the snap

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

Up til this point, it's not been possible to update from one major/minor version of EdgeX due to the use of sequential epochs in snapcraft.yaml. This policy may be changed with the Hanoi. If so, when someone attempts to refresh from the geneva snap and they have support-logging enabled, the refresh/update should fail to install with an appropriate error message (**note**, this has not been implemented in this PR). This is similar to what would happen if someone were to update from geneva and they had mongo or the old support rules-engine enabled.

## Are there any new imports or modules? If so, what are they used for and why?
No

### CODE REVIEW

**Comment by tonyespy** [comment]
> > Up til this point, it's not been possible to update from one major/minor version of EdgeX due to the use of sequential epochs in snapcraft.yaml. This policy may be changed with the Hanoi, however if someone attempts to refresh from the geneva snap and they have support-logging enabled, the refresh/update will fail to install. This is similar to what would happen if someone were to update from geneva and they had mongo or the old support rules-engine enabled.
> 
> To be clear, what this PR is doing is removing the service so that it doesn't attempt to be started, so before this PR if someone refreshed from geneva/stable (or latest/stable) to edge, they would get broken because support-logging is broken on edge, but now after this PR if they did that same refresh it wouldn't be broken at least due to support-logging because now support-logging just disappears correct?

I'm not sure what you mean by "support-logging is broken on edge"? We haven't yet updated the epoch for Hanoi, so technically it's possible for someone to refresh from geneva to latest/edge, and it they had support-logging enabled, then it will be gone after the refresh, and no errors will be generated.

When we landed the changes for the legacy rules-engine and mongo, we didn't add logic to the refresh hook to fail if either were being used, because we'd been using epochs to prevent refreshes between releases. As such, I'll adjust this PR to bump the epoch to 4 to prevent refreshes. If we decide to support refreshes from geneva, then we'll need to ensure that the refresh hook as the proper logic to detect current usage of deprecated services.

**Comment by anonymouse64** [comment]
@tonyespy can you clarify what you mean by this in the PR description: 

> however if someone attempts to refresh from the geneva snap and they have support-logging enabled, the refresh/update will fail to install

I'm confused about this statement since in your response just now you also say:

> if they had support-logging enabled, then it will be gone after the refresh, and no errors will be generated.

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -16,10 +16,9 @@ icon: snap/local/assets/edgex-snap-icon.png
 # Different epochs prevent refreshes between major versions of EdgeX due
 # to continued configuration changes.
 #
-# delhi: 0, edinburgh: 1, fuji: 2, geneva: 3
-epoch: 3
+# delhi: 0, edinburgh: 1, fuji: 2, geneva: 3, hanoi: 4
+epoch: 4
 
-# TODO: add armhf here when that's supported
```
Comment (anonymouse64) [inline snap/snapcraft.yaml:22]: why remove this TODO ?

Comment (tonyespy) [reply snap/snapcraft.yaml:22]: Because we're never going  to support it, and I got tired of seeing this comment everytime I edited snapcraft.yaml. ;)-

**Review by anonymouse64 — DISMISSED** [review]

**Comment by anonymouse64** [comment]
Built and installed on ubuntu 20.04 classic, all services came up and support-logging was gone, so lgtm (meant for this to be a part of the previous review)

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**